src/front/zsharp/term.rs

```diff
@@ -772,7 +769,7 @@ pub fn array_store(array: T, idx: T, val: T) -> Result<T, String> {
 }
 
 fn ir_array<I: IntoIterator<Item = Term>>(sort: Sort, elems: I) -> Term {
-    let mut values = BTreeMap::new();
+    let mut values = HashMap::new();
```

Comment:
So I assume the speedup here is in the case that there are a lot of constants in the array?

Reply:
Yeah, exactly. Large constant arrays were a big speed issue previously.